### DISCUSSION TITLE
Stitching Multiple navGens

### DISCUSSION THREAD

**Discussion by niko-dellic** [isaac-mason/recast-navigation-js#450] [Q&A]
Is it possible to generate multiple navMeshes and then stitch them together into one navMesh?

**Comment by rob-myers** [comment]
It's possible to add/remove/replace "tiles" in a tiled navMesh, which amounts to the same thing:

```ts
const navMesh = new NavMesh()
navMesh.initTiled(navMeshParams)
// ...
navMesh.removeTile(...)
navMesh.addTile(...)
```

There's a [dynamic navmesh demo here](https://sketches.isaacmason.com/sketch/recast-navigation/dynamic-tiled-navmesh), which I am currently trying to replicate.

**Reply by niko-dellic** [reply]
This is a good start, although I am getting issues when trying to add a tile to my tiledNavMesh:
Failed to add tile at (38,31,0): -2147483647. I understand this means DT_FAILURE.

Essentially I am trying to combine multiple navMeshes. Imagine if the model below was split down the middle and both sides generated a NavMesh (while this is simplified, imagine doing this to handle large files).
 
![split](https://github.com/user-attachments/assets/13ee318a-8cb8-40f8-a22f-b0a4e57658f1)

My hope is to:
1. load multiple .bin (uint8Array) files of downloaded navMesh results
2. load the first element using importNavMesh from core
3. stitch the following uint8Array elements via AddTile

i've gotten DT_FAILURE in cases where I have used both a duplicate of the same navMesh (for testing) and in cases where they are different navMeshes. In both cases, they are generated with the same settings.

Any further advice would be greatly appreciated!
Niko

**Reply by rob-myers** [reply]
Let's say the 1st geometry is one half and the 2nd geometry is the other half.

Are you removing all the tiles touching the 2nd geometry?
Are you including both geometries when re/adding tiles? 

You could visualise the tiles in XZ plane using three.js [GridHelper](https://threejs.org/docs/#api/en/helpers/GridHelper).

**Comment by rob-myers** [comment]
I'm not sure whether non-tiled navmeshes amount to a "directed graph over triangles".
Maybe that's what you're asking?
